### ISSUE TITLE
Possible bug in copy(shallow=False)

### ISSUE THREAD

**Issue by hmstepanek** [data-8/datascience#453]
# Problem
Table.copy has a parameter option shallow. By default, shallow is set to False.  One would expect that by default this would preform what's commonly known as a "deepcopy". Meaning, the copied Table's objects contain no references to the original and they each can be changed independently without changing the other. However, this is in fact not the case. Both Table.copy(shallow=True) and Table.copy(shallow=False) are forms of a shallow copy.

## Expected Behavior
The expected behavior of a shallow=False copy would be that when updating the original_table, table_copy is unaffected.
```
>>> value = {"foo"}
>>> original_table = Table().with_columns(
...    "A", make_array(1, 2, 3),
...    "B", make_array(value, {"foo", "bar"}, {"foo"}),
... )
>>> original_table
A    | B
1    | {'foo'}
2    | {'foo', 'bar'}
3    | {'foo'}
>>> table_copy = original_table.copy()
>>> table_copy
A    | B
1    | {'foo'}
2    | {'foo', 'bar'}
3    | {'foo'}
>>> value.add("bar")
>>> original_table
A    | B
1    | {'foo', 'bar'}
2    | {'foo', 'bar'}
3    | {'foo'}
>>> table_copy
A    | B
1    | {'foo'}
2    | {'foo', 'bar'}
3    | {'foo'}

```
## Actual Behavior
The actual behavior is that when updating the original_table, table_copy updates as well.
```
>>> value = {"foo"}
>>> original_table = Table().with_columns(
...    "A", make_array(1, 2, 3),
...    "B", make_array(value, {"foo", "bar"}, {"foo"}),
... )
>>> original_table
A    | B
1    | {'foo'}
2    | {'foo', 'bar'}
3    | {'foo'}
>>> table_copy = original_table.copy()
>>> table_copy
A    | B
1    | {'foo'}
2    | {'foo', 'bar'}
3    | {'foo'}
>>> value.add("bar")
>>> original_table
A    | B
1    | {'foo', 'bar'}
2    | {'foo', 'bar'}
3    | {'foo'}
>>> table_copy
A    | B
1    | {'foo', 'bar'}
2    | {'foo', 'bar'}
3    | {'foo'}

```

The shallow switch behavior is implemented here:
```
            if shallow:
                column = self[label]
            else:
                column = np.copy(self[label])
            self._add_column_and_format(table, label, column)
```
Note that np.copy is not a deepcopy, according to the [docs](https://numpy.org/doc/stable/reference/generated/numpy.copy.html).

> Note that np.copy is a shallow copy and will not copy object elements within arrays. This is mainly important for arrays containing Python objects. The new array will contain the same object which may lead to surprises if that object can be modified (is mutable):

# Possible Bug
The question here is: is Table.copy(shallow=False) expected to be a deepcopy? If so, there is a bug in the implementation of Table.copy.

# Proposed fix
Use [copy.deepcopy](https://docs.python.org/dev/library/copy.html#copy.deepcopy) instead:
```
            if shallow:
                column = self[label]
            else:
                column = copy.deepcopy(self[label])
            self._add_column_and_format(table, label, column)
```

**Comment by adnanhemani** [comment]
So I can't comment on the original thought of the library creators - but I agree with your assessment that perhaps this is a bug and that `shallow=False` should be a deep copy rather than using `np.copy`. Please go ahead and create a PR with this code fix (seems like you know what's needed) and I'll review. Thanks!

**Comment by adnanhemani** [comment]
Merged in #454 . Thanks for the PR :)
